### PR TITLE
replace active by enabled for consistency in plugins

### PR DESCRIPTION
**Pull request by vieux**
depends on https://github.com/docker/engine-api/pull/351

ping @anusha-ragunathan @tiborvass

### CODE REVIEW

**Comment by tiborvass** [comment]
I wonder if it should be `State: Enabled` instead. cc @stevvooe wdyt?

File: plugin/manager.go [hunk 1]
```diff
@@ -364,7 +364,7 @@ func (pm *Manager) initPlugin(p *plugin) error {
 }
 
 func (pm *Manager) remove(p *plugin) error {
-	if p.PluginObj.Active {
+	if p.PluginObj.Enabled {
 		return fmt.Errorf("plugin %s is active", p.Name())
```
Comment (anusha-ragunathan) [inline plugin/manager.go]: Should be "plugin %s is enabled"

File: plugin/manager_linux.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func (pm *Manager) Shutdown() {
 
 	pm.shutdown = true
 	for _, p := range pm.plugins {
-		if pm.liveRestore && p.PluginObj.Active {
+		if pm.liveRestore && p.PluginObj.Enabled {
 			logrus.Debug("Plugin active when liveRestore is set, skipping shutdown")
```
Comment (anusha-ragunathan) [inline plugin/manager_linux.go]: "Plugin active" -> "Plugin enabled"

**Comment by stevvooe** [comment]
@tiborvass What other states are there?

**Comment by tiborvass** [comment]
@stevvooe none other at this point.

File: plugin/manager.go (hunk 1 above)
Comment (vieux) [inline plugin/manager.go]: done, thanks.

File: plugin/manager_linux.go (hunk 2 above)
Comment (vieux) [inline plugin/manager_linux.go]: done, thanks.

**Comment by stevvooe** [comment]
@tiborvass Ultimately, a state model is probably better here but an `Enabled` flag is fine to use with something like that over time and may even be wholly separate.

**Comment by thaJeztah** [comment]
"Enabled" is definitely something I'd prefer over "Active" (mentioned this when reviewing the feature), as it's consistent with `docker plugin enable/disable`). For consistency with `docked ps`, a `STATE` or `STATUS` column would make sense

(I just noticed we're not consistent - `node ps` uses `STATE`, but `ps` uses `STATUS`)

**Comment by stevvooe** [comment]
As for the API field, `Enabled` seems the right way to go.

We should make the UX decision separately. How do we intend to display the different lifecycle states of a plugin. We have at least "available", "installed" and "running". Aren't there others?

**Comment by vieux** [comment]
this PR is on hold, there is too much changes out of scope of this PR  in engine-api , therefore, I cannot vendor.

**Comment by anusha-ragunathan** [comment]
@stevvooe : There are only 2 states of a plugin. enabled and disabled. 
Re: "We have at least "available", "installed" and "running"
- When plugin is "installed", it is enabled by default. If user wants to override this behaviour, they can by install and disable it.
- An enabled plugin is "running". A disabled plugin is not.

**Comment by stevvooe** [comment]
@anusha-ragunathan Go play with any text editor or IDE: there are usually at least three states. Usually, you can search, install, enable and disable a plugin.

I think you can argue that "installed" and "available" are the same thing, if you don't have a remote plugin registry. In that case, enabled/disabled is sufficient but you lose the lifecycle-state isomorphism that using another model may provide.

**Comment by anusha-ragunathan** [comment]
@stevvooe : There's no universal rule about plugin states. Browser plugins for example are installed and enabled at the same time. You can optionally disable it. Mobile apps are installed and enabled at the same time and can be optionally disabled.

**Comment by stevvooe** [comment]
> There's no universal rule about plugin states.

Probably not. But there are models that developers are familiar with. Browser plugins and mobile apps aren't a good model, since those are closer to end users. Editor models are closer to what Docker might want to use, since users are more likely to want to modify or understand plugin contents.

Enabled/Disabled is probably sufficient.

**Comment by stevvooe** [comment]
LGTM

**Comment by thaJeztah** [comment]
Can plugins fail to start? And would we like to add more information in future? Thinking of

```
NAME                  TAG                 Status
tiborvass/no-remove   latest              Enabled 59 minutes ago
vieux/sshfs           latest              Disabled
thajeztah/foo         latest              Exited 30 seconds ago
```

TL;DR I _do_ think this is more consistent with the command (`plugin enable/disable`), just wondering if we should be consistent with `docker ps` as well

**Comment by cpuguy83** [comment]
@vieux engine-api should be ok now.

**Comment by anusha-ragunathan** [comment]
@thaJeztah : If a plugin fails to start, it is in disabled state and not restarted by the plugin restartManager.

**Comment by vieux** [comment]
it should be good now

**Comment by vdemeester** [comment]
LGTM 🐸

**Comment by tiborvass** [comment]
LGTM
